File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit=0,today=0,least=prices[0];
        for(int i=0;i<prices.size();i++)
        {   
            if(prices[i]<least)
                least = prices[i];
            today=prices[i]-least;
            if(profit<today)
                profit=today;
        }
        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min=prices[0], profit=0,max_profit=0;
        for(int i=0;i<prices.length;i++){
            if(prices[i]<min)
                min= prices[i];
            profit = prices[i]-min;
            if(profit>max_profit){
                max_profit= profit;
            }
        }
        return max_profit;
    }
}